### DISCUSSION TITLE
Create a Node and attach script upon drag and drop of a .gd file

### DISCUSSION THREAD

**Discussion by Supamiu** [godotengine/godot-proposals#8283] [Editor]
## The problem
When creating composition-oriented elements in the editor, let's say logic bricks that you'd like to use to compose items, you have to create a gdscript and a packed scene in order ot be able to use drag and drop to compose the elements, taking this structure as example (where each node also updates its name to make it easier to read the effect of the item):
![image](https://github.com/godotengine/godot-proposals/assets/11519203/d94b2cca-1cfe-4acd-8eb8-f6eeddb91ebd)

## Current approach
At the moment, each `Node` of the screenshot shown above is just a `Node` with a gdscript attached to it. Creating these nodes can be made using the "right click => 'Add Child Node' => pick 'Node' => attach desired behavior script to it" approach.
This means 4 steps to add my script to a new node, becoming a bit much when more complex behaviors are going to be composed with more logic components.

To make this easier to compose, I created a `.tscn` file for each component that can be used in this context, each `.tscn` being just a `Node` with the component's script attached to it.

## Proposal
As I believe composition is a core concept of the engine (and a quite powerful one in videogames design overall), I think it's make sense to add a way to create a node on the fly when dropping a script.

My current idea would be to allow this through a modifier keypress (like Ctrl, Alt or Shift) while the drop is being made, Editor would then create a node using the type that the script extends as base type and attach the script to it.

**Comment by theraot** [comment]
This feature already exists: drag the script with CTRL and it will make a child Node with the script attached (if the script extends a Node type or derived).

You are confirming once more that it is a hard to discover feature. I did the same once before: https://github.com/godotengine/godot-proposals/discussions/7211

**Reply by Supamiu** [reply]
You're so damn right.

First of all thanks a lot for making me discover this. But also I was discussing the fact that shortcuts are quite bad by design because they're hard to discover, as a gaming tool developer myself. It's funny to see that it also affected me few days later 😄
